src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java

```diff
@@ -53,59 +62,101 @@
 	@Deprecated
 	private transient String priorityParameterName;
 	private List<LockableResource> resources;
+	private List<LockableResource> stratosResources;
+	private List<LockableResource> localResources;
 
-	private StratOSControllerAPI controllerAPI;
+	private transient StratOSControllerAPI controllerAPI;
+	
+	private String stratosURL = null;
+	private String username = null;
+	private String password = null;
 
 	/**
 	 * Only used when this lockable resource is tried to be locked by {@link LockStep},
 	 * otherwise (freestyle builds) regular Jenkins queue is used.
 	 */
 	private List<QueuedContextStruct> queuedContexts = new ArrayList<QueuedContextStruct>();
 
-	public LockableResourcesManager() {
+public LockableResourcesManager() {
+		
+		// load the config before we do anything
+		load();
+
 		StratOSClientConfiguration config;
-		try {
-			config = Builders.config().url(new URL("http://localhost:8880"))
-			                                                    .username("admin")
-			                                                    .password("admin")
-			                                                    .build();
+		String url = getStratosURL();
+		String username = getUsername();
+		String password = getPassword();
+		
+		if (url != null && username != null && password != null){
 			try {
-				controllerAPI = ClientFactory.getControllerAPI(config);
-			} catch (StratOSRESTClientConfigurationException e) {
+				config = com.seastreet.client.config.Builders.config().url(new URL(url)).username(username).password(password).build();
+				try {
+					controllerAPI = ClientFactory.getControllerAPI(config);
+				} catch (StratOSRESTClientConfigurationException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
```

Comment:
What is the proper way to handle exceptions in a Jenkins plugin? The plugin can't function properly if the server isn't reachable. Make sure you test this case.